pydis_site/apps/api/migrations/0094_migrate_mailing_listdata.py

```diff
@@ -0,0 +1,45 @@
+# Generated by Django 5.0 on 2023-12-13 07:03
+
+from django.db import migrations
+
+
+def migrate_mailing_lists_into_new_model(apps, schema_editor):
+    """Move the bot's mailing list information from the BotSetting to the new MailingList model."""
+
+    BotSetting = apps.get_model('api', 'BotSetting')
+    MailingList = apps.get_model('api', 'MailingList')
+    MailingListSeenItem = apps.get_model('api', 'MailingListSeenItem')
+    try:
+        setting = BotSetting.objects.get(name='news')
+    except BotSetting.DoesNotExist:
+        return
+
+    # Field format:
+    # {
+    #   "pep": [
+    #     "644",
+    #     "8102",
+    #     ...
+    for list_name, item_hashes in setting.data.items():
+        (mailing_list, _created) = MailingList.objects.get_or_create(name=list_name)
+        MailingListSeenItem.objects.bulk_create(
+            MailingListSeenItem(list=mailing_list, hash=item_hash)
+            for item_hash in item_hashes
+        )
+
+    setting.delete()
+
+
+def do_nothing(*args, **kwargs):
```

Comment:
The `reverse_code` param defaults to None, so no need to have the `do_nothing`  method def.

Reply:
Oh, I actually left this in by accident. I had to add this to allow me to roll back the migration properly (if you don't specify the reverse, it will error out as being irreversible). I've removed it now for The Big Squash 2023